bundles/org.eclipse.equinox.p2.metadata/.settings/.api_filters

```diff
@@ -8,4 +8,12 @@
             </message_arguments>
         </filter>
     </resource>
+    <resource path="src/org/eclipse/equinox/p2/query/IQueryResult.java" type="org.eclipse.equinox.p2.query.IQueryResult">
```

Comment:
IQueryResult is not marked with `@noimplement/@noextend` but I wonder if clients are really supposed to create own implementations?

Reply:
Tycho has some implementations of `IQueryResult`, I don't see why one should not allow this?

Reply:
People do what they're not supposed to do, so if they're allowed, one can assume that someone will to it.  😱 

What about a default method that does toSet().stream() so that by default there is always an implementation?

Reply:
OK with custom queryable this makes sense.
Tycho even already has a stream() method in its own implementation, but it only needs to be made public when depending on the next Eclipse Release, so I think this is fine.
https://github.com/eclipse-tycho/tycho/blob/master/p2-maven-plugin/src/main/java/org/eclipse/tycho/p2maven/ListQueryable.java#L80-L83

Reply:
`IQueryable` already implements `Iterable` so using the iterator for default implementations seems a better choice to prevent creation of an intermediate set.

Reply:
> What about a default method that does toSet().stream() so that by default there is always an implementation?

That's possible. But as Christoph suggested it could use the `Spliterator` of the `IQueryResult`, like it is currently done in `QueryResult`.

`Iterable` itself unfortunately does not provide a stream() method (if it would this PR would be pointless).

Reply:
> > What about a default method that does toSet().stream() so that by default there is always an implementation?
> 
> That's possible. But as Christoph suggested it could use the `Spliterator` of the `IQueryResult`, like it is currently done in `QueryResult`.

Done.